src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/SlhDsaImplementation.OpenSsl.cs

```diff
@@ -0,0 +1,92 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Buffers;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+
+namespace System.Security.Cryptography
+{
+    internal sealed partial class SlhDsaImplementation : SlhDsa
+    {
+        private SafeEvpPKeyHandle _key = null!;
+
+        private SlhDsaImplementation(SlhDsaAlgorithm algorithm, SafeEvpPKeyHandle key)
+            : base(algorithm)
+        {
+            Debug.Assert(key is not null);
+            Debug.Assert(!key.IsInvalid);
+            Debug.Assert(SupportsAny());
+
+            _key = key;
+        }
+
+        internal static partial bool SupportsAny() =>
+            Interop.Crypto.EvpPKeySlhDsaAlgs.SlhDsaSha2_128s != null ||
+            Interop.Crypto.EvpPKeySlhDsaAlgs.SlhDsaShake128s != null ||
+            Interop.Crypto.EvpPKeySlhDsaAlgs.SlhDsaSha2_128f != null ||
+            Interop.Crypto.EvpPKeySlhDsaAlgs.SlhDsaShake128f != null ||
+            Interop.Crypto.EvpPKeySlhDsaAlgs.SlhDsaSha2_192s != null ||
+            Interop.Crypto.EvpPKeySlhDsaAlgs.SlhDsaShake192s != null ||
+            Interop.Crypto.EvpPKeySlhDsaAlgs.SlhDsaSha2_192f != null ||
+            Interop.Crypto.EvpPKeySlhDsaAlgs.SlhDsaShake192f != null ||
+            Interop.Crypto.EvpPKeySlhDsaAlgs.SlhDsaSha2_256s != null ||
+            Interop.Crypto.EvpPKeySlhDsaAlgs.SlhDsaShake256s != null ||
+            Interop.Crypto.EvpPKeySlhDsaAlgs.SlhDsaSha2_256f != null ||
+            Interop.Crypto.EvpPKeySlhDsaAlgs.SlhDsaShake256f != null;
```

Comment:
This is... perfectly true.  But if we're actually anticipating any of them having a different nullarity than the rest we probably want to add a static `Supports(Algorithm)`.

Reply:
OpenSSL implemented them all in one shot but I guess it'll depend on what Windows exposes. I suppose it also depends on if we think that more algos will be added to this family. And also symmetry with ML-DSA and Composite ML-DSA (the latter of which might not have support for all algos on some platforms).

Reply:
Composite ML-DSA will definitely have to express what algorithms are supported, since we definitely won't support all of them.

OpenSSL 3.5 does not define Composite ML-DSA as an inbuilt algorithm, so we'll be doing the "compositing" in the .NET layer.  Since we don't have support for EdDSA, the EdDSA-based-composites will not be supported on Linux.  (They also won't be supported on Windows, since Windows doesn't support EdDSA at all, part of why we don't support EdDSA :smile:).

The point here is really that this has a lot of `|| bar != null`, but they're decided by the very first one.  I had similar thoughts on ML-*, but they just had three, so it was easier to ignore.  Here we're doing "much" more work to conclude `false`.